### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by dhruvin25799**
## Added Profile Page
Live: https://magenta-taffy-45e62a.netlify.app/

## Functionality to check : 

- [ ] Test Email : itachiuchiha25799@gmail.com Password : 1234567890
- [ ] Profile page for your profile
- [ ] You can visit other profiles by clicking on their icon

## Files to check : 

- [ ] src/components/ProfileSidebar/ProfileSidebar.jsx
- [ ] src/helpers/getFunctions.js
- [ ] src/helpers/postFunctions.js
- [ ] src/pages/Home/Home.jsx
- [ ] src/pages/Profile/Profile.jsx
- [ ] src/store/post-slice.js
- [ ] src/store/profile-slice.js

### CODE REVIEW

**Review by iHarryD — APPROVED** [review]
Everything looks good. Code is clean and easy to read. Good to merge.

File: src/App.js [hunk 1]
```diff
@@ -14,20 +11,30 @@ function App() {
...
+                  <Profile />
+                </RequiresAuth>
+              }
+            />
+            <Route path="/login" element={<Login />} />
+            <Route path="/signup" element={<SignUp />} />
+          </Routes>
+        </div>
       </div>
     </>
```
Comment (theVivek-kumar) [inline src/App.js:39]: you can use outlet 
 <Outlet> element is used as a placeholder. In this case an <Outlet> enables the Users component to render its child routes.

File: src/components/Avatar/Avatar.module.css [hunk 2]
```diff
@@ -1,8 +1,17 @@
-.avatar-container{
-    display: flex;
-    flex-flow: column;
-    align-items: center;
-    width: 50px;
```
Comment (theVivek-kumar) [inline src/components/Avatar/Avatar.module.css:5]: use rem

File: src/components/Avatar/Avatar.module.css [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
-    cursor: pointer;
-}
\ No newline at end of file
+.avatar-container {
+  display: flex;
+  flex-flow: column;
+  align-items: center;
+  width: 50px;
+  height:70px;
+  cursor: pointer;
```
Comment (theVivek-kumar) [inline src/components/Avatar/Avatar.module.css:7]: here use rem

File: src/components/LoadingSpinner/LoadingSpinner.module.css [hunk 4]
```diff
@@ -0,0 +1,16 @@
+@keyframes spinner {
+  0% {
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+}
+.loading-spinner {
+  width: 50px;
```
Comment (theVivek-kumar) [inline src/components/LoadingSpinner/LoadingSpinner.module.css:10]: here use rem and color variable

File: src/components/Post/Post.module.css [hunk 5]
```diff
@@ -1,51 +1,61 @@
...
-.post-header{
-    display: flex;
-    flex-flow: row;
-    align-items: center;
+.post-card a {
+  text-decoration: none;
+  color: var(--COLOR-text);
 }
 
-.author-icon{
```
Comment (theVivek-kumar) [inline src/components/Post/Post.module.css:16]: here use rem

File: src/helpers/getFunctions.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+export const getUserPosts = async (username) => {
+  const response = await axios.get("/api/posts/user/" + username);
+  return response.data;
+};
+
+export const getSingleUser = async (username) => {
+  const response = await axios.get("/api/users/"+ username);
+  return response.data;
+};
```
Comment (theVivek-kumar) [inline src/helpers/getFunctions.js:21]: please provide live link for UI or give ss

File: src/pages/Profile/Profile.module.css [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+.profile-header {
+  display: flex;
+  flex-flow: column;
+  margin-left: 10rem;
+}
+
+.profile-info {
+  display: flex;
+  flex-flow: row;
+  gap: 5rem;
```
Comment (theVivek-kumar) [inline src/pages/Profile/Profile.module.css:22]: clean code

**Review by theVivek-kumar — COMMENTED** [review]
only suggestion is use rem or try outlet

File: src/helpers/getFunctions.js (hunk 6 above)
Comment (dhruvin25799) [reply src/helpers/getFunctions.js:21]: The live link is in the description of every PR.

**Pull request closed without merging**